docs/tutorial/upgrade-nfs-server.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+# Copyright 2021 The OpenEBS Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -e
+
+print_help() {
+cat << EOF
+Usage:
+    $0 <options> IMAGE_TAG
+
+Example:
+    $0 -n nfs-ns 0.7.1
+
+IMAGE_TAG is required to execute this script.
+By default, this script uses 'openebs' namespace to search nfs-server deployment.
+If you have used different namespace for nfs-server deployment then you must provide
+the namespace using option(-n).
+EOF
+}
+
+# NFS_SERVER_NS represent the namespace for nfs server deployment
+# By default, it is set to 'openebs'.
+NFS_SERVER_NS=openebs
+
+# IMAGE_TAG represent the version for nfs server image
+IMAGE_TAG=
+
+# list_deployment list nfs-server deployment in NFS_SERVER_NS namespace
+list_deployment() {
+    local -n deploy=$1
+    deploy=$(kubectl get deployment -n ${NFS_SERVER_NS} -l openebs.io/nfs-server --no-headers -o custom-columns=:.metadata.name)
+}
+
+# upgrade_deployment patch the given deployment in NFS_SERVER_NS with image-tag IMAGE_TAG
+upgrade_deployment() {
+    deploymentName=$1
+
+    existingImage=$(kubectl get deployment -n openebs ${deploymentName} -o jsonpath='{.spec.template.spec.containers[0].image}')
+
+    repo=${existingImage%:*}
+    newImage=${repo}:${IMAGE_TAG}
+    patchJson="{\"spec\": {\"template\": {\"spec\": {\"containers\" : [{\"name\" : \"nfs-server\", \"image\" : \"${newImage}\"}]}}}}"
+
+    kubectl patch deploy -n ${NFS_SERVER_NS} ${deploymentName} -p "${patchJson}" > /dev/null
+    exitCode=$?
+    if [ $exitCode -ne 0 ]; then
```

Comment:
What happens in the following case:
- There are 2 nfs server deployments
- First run successfully upgraded 1 of them to latest images and script exited due to network issues or some other case
- User tries to upgrade again the remaining ones
- Will the first one be skipped or does the patch command return an error and not continue with upgrading next nfs server. 

Also, can we have an optional parameter that says if user should upgrade only one PVC instead of all PVCs..

Reply:
In case of a partial upgrade, the second execution of the upgrade script will skip the upgrade for already upgraded deployment and proceed with the upgrade for non-upgraded nfs-server deployments.

I've updated the script to upgrade specific nfs-server deployment. Also updated the upgrade document with the command for the same.